src/static-build-loader/loader.ts

```diff
@@ -0,0 +1,162 @@
+import getFeatures from './getFeatures';
+import { LoaderContext, RawSourceMap } from 'webpack/lib/webpack';
+import * as recast from 'recast';
+
+const { getOptions } = require('loader-utils');
+const types = recast.types;
+const namedTypes = types.namedTypes;
+const builders = types.builders;
+const compose = require('recast/lib/util').composeSourceMaps;
+
+/**
+ * A map of features that should be statically replaced in the code
+ */
+export interface StaticHasFeatures {
+	[feature: string]: boolean;
+}
+
+const HAS_MID = /\/has$/;
+const HAS_PRAGMA = /^\s*(!?)\s*has\s*\(["']([^'"]+)['"]\)\s*$/;
+const HAS_MID_SNIFF = /require\(["'][^,"']+\/has["']\)/;
+const HAS_PRAGMA_SNIFF = /^\s*['"]\s*(!?)\s*has\s*\(["']([^'"]+)['"]\)\s*['"]\s*;\s*$/m;
+
+/**
+ * Checks code for usage of has pragmas or other calls to @dojo/has and optimizes them out based on the flags or
+ * feature sets specified statically. This loader should act on JavaScript, so it should run after the compiler
+ * if using TypeScript
+ * @param content The JavaScript code to optimize
+ * @param sourceMap Optional Source map for the code. If provided it will be updated to reflect the optimizations made
+ */
+export default function loader(this: LoaderContext, content: string, sourceMap?: RawSourceMap): string | void {
+	if (!(HAS_MID_SNIFF.test(content) || HAS_PRAGMA_SNIFF.test(content))) {
+		return content;
```

Comment:
don't we also need to check and return a sourcemap if it exists? otherwise I think it will get blown away?

Reply:
yeah thats a good idea 👍, i do that in https://github.com/matt-gadd/umd-compat-loader/blob/master/test/index.js#L56 (i also physically assert the sourcemap content but thats probably overkill)